src/QsCompiler/Tests.Compiler/TestCases/TypeParameterResolution.qs

```diff
@@ -157,3 +157,110 @@ namespace Microsoft.Quantum.Testing.TypeParameterResolution {
 
 // Foo.A -> Int
 // Bar.B -> Int
+
+// =================================
+
+// Basic Entry Point
+namespace Microsoft.Quantum.Testing.TypeParameterResolution {
+
+    @ EntryPoint()
+    operation Main() : Unit {
+        Foo();
+        Bar();
+    }
+
+    operation Foo() : Unit { }
+
+    operation Bar() : Unit {
+        Baz();
+    }
+
+    operation Baz() : Unit { }
+}
+
+// =================================
+
+// Unrelated To Entry Point
+namespace Microsoft.Quantum.Testing.TypeParameterResolution {
+
+    @ EntryPoint()
+    operation Main() : Unit {
+        Foo();
+    }
+
+    operation Foo() : Unit { }
+
+    operation Bar() : Unit {
+        Baz();
+    }
+
+    operation Baz() : Unit { }
+}
+
+// =================================
+
+// Separated From Entry Point By Specialization
+namespace Microsoft.Quantum.Testing.TypeParameterResolution {
+
+    @ EntryPoint()
+    operation Main() : Unit {
+        Adjoint Foo();
+    }
+
+    operation Foo() : Unit is Adj {
+        body(...) {
+            Bar();
+        }
+
+        adjoint(...) { }
+    }
+
+    operation Bar() : Unit { }
+}
+
+// =================================
+
+// Not Called With Entry Point
```

Comment:
>Not Called With Entry Point [](start = 3, length = 27)

Isn't this test checking the same case as "Unrelated To Entry Point"?

Reply:
The two Not Called tests cover how the call graph walker will respond to an operation that is never called. This is subtly different from a pair of operations that are called but are not descendants of an Entry Point. The expected behavior for these two situations is the same, but the situations are distinct.

Reply:
My understanding is that `Bar` in "Unrelated To Entry Point" is not called, similar to `NotCalled` here. Thus, "Unrelated" test covers strictly bigger subset of functionality compared to this one, no? 

---
In reply to: [439689809](https://github.com/microsoft/qsharp-compiler/pull/454#discussion_r439689809) [](ancestors = 439689809)

Reply:
Because `NotCalled` is never called by anything nor calls anything, it represents a different situation from `Bar` which calls things and would be in the call graph except that it is not related to an Entry Point.

Reply:
Do I understand it correctly, that "NotCalled" won't be in the callgraph even if the entry point wasn't specified, because it has no dependencies in or out? Do you have a test for this among the no-entry-point cases? If this understanding is correct, this test seems to be even more superfluous. That is, unless the logic of building the callgraph from entry points has nothing to do with the logic of building the full graph, in which case I'd expect most of the tests to be duplicated.
However, I'd add a test case with NotCalled marked as entry point to highlight that in this situation the callgraph should remain empty (again, assuming my understanding above is correct).

Reply:
Yes your understanding is correct. This is a good suggestion for a test, and I have now added such a test.